### DISCUSSION TITLE
cesium-nodekit: support of self hosted Cesium ION ?

### DISCUSSION THREAD

**Discussion by remoe** [pelicanmapping/osgearth#2612] [Support Q&A] [answered]
Does support cesium-nodekit a self hosted Cesium ION with asset id's? It needs accessToken per layer and a server configuration entry for the cesium api server in the earth file?

In CesiumJS (javascript) one can add it as follow:

```js
const resource = await IonResource.fromAssetId(assetID, {
  accessToken: '<TOKEN>',
  server: 'https://example.ion'
});

const terrainProvider = CesiumTerrainProvider.fromUrl(resource);

```

**Comment by jasonbeverage** [comment, marked as answer]
We don't currently support self hosted Cesium Ion but we could.  I've added an issue to track this at https://github.com/gwaldron/osgearth/issues/2613 to track this.
